src/test/java/io/confluent/connect/jdbc/source/integration/MSSQLDateTimeIT.java

```diff
@@ -0,0 +1,314 @@
+                CONNECTOR_NAME,
+                Math.min(KAFKA_TOPICS.size(), TASKS_MAX),
+                "failed to verify that tasks have failed"
+        );
+
+        deleteTable();
+    }
+
+
+    @Test
+    public void verifyDateTimeTSAndIncrModeKillsTasks() throws Exception {
+        // Setup up props for the source connector
+        props = configProperties(false);
+        createDateTimeTable("DATETIME", "VARCHAR(255)", false);
+        ZoneId utc = ZoneId.of("UTC");
+        // Create topic in Kafka
+        KAFKA_TOPICS.forEach(topic -> connect.kafka().createTopic(topic, 1));
+
+        // Configure source connector
+        configureAndWaitForConnector();
+
+        java.sql.Timestamp t = java.sql.Timestamp.from(
+                ZonedDateTime.of(2016, 12, 8, 19, 34, 56, 2000000, utc).toInstant()
+        );
+
+        writeSingleRowWithTimestamp(t, true, false);
+
+        connect.assertions().assertConnectorIsRunningAndTasksHaveFailed(
+                CONNECTOR_NAME,
+                Math.min(KAFKA_TOPICS.size(), TASKS_MAX),
+                "failed to verify that tasks have failed"
+        );
+
+        deleteTable();
+    }
+
+
+    /**
+     * Verify that after a number of poll intervals the number of records in topic
+     * still equal to the number of records written to the MSSQL Server instance.
+     *
+     * This test is for a datetime to datetime2 comparison error in MSSQL Server (2016 and after).
+     * Datetime is converted recursively to a higer precision (eg- 3.33 MS would be 3.33333 ),
+     * while java.sql.Timestamp converts by tacking on 0s (3.33000).
+     */
+
+    @Test
+    public void verifyDATETIME2DoesNotKillTask() throws Exception {
+        props = configProperties(true);
+        // And verify that datetime in a non-timestamp column does not kill task.
+        createDateTimeTable("DATETIME2", "DATETIME", true);
+        ZoneId utc = ZoneId.of("UTC");
+        // Create topic in Kafka
+        KAFKA_TOPICS.forEach(topic -> connect.kafka().createTopic(topic, 1));
+
+        // Configure source connector
+        configureAndWaitForConnector();
+
+        java.sql.Timestamp t = java.sql.Timestamp.from(
+                ZonedDateTime.of(2016, 12, 8, 19, 34, 56, 2000000, utc).toInstant()
+        );
+
+        writeSingleRowWithTimestamp(t, false, true);
+
+        Thread.sleep(Duration.ofSeconds(30).toMillis());
+        for (String topic: KAFKA_TOPICS) {
+            ConsumerRecords<byte[], byte[]> records = connect.kafka().consume(
+                    NUM_RECORDS_PRODUCED,
+                    CONSUME_MAX_DURATION_MS,
+                    topic);
+            //Assert that records in topic == NUM_RECORDS_PRODUCED
+            assertEquals(NUM_RECORDS_PRODUCED, records.count());
+
+        }
+        // clean up table
+        deleteTable();
+    }
+
+    private Map<String, String> configProperties(boolean timestampModeOnly) {
```

Comment:
Let's pass the actual mode instead of the bool. It looks goofy when you call it with `configProperties(true)`